test/integration/network/dual_stack_test.go

```diff
@@ -0,0 +1,336 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"fmt"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-base/featuregate"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	"k8s.io/kubernetes/cmd/kube-apiserver/app/options"
+	"k8s.io/kubernetes/pkg/features"
+	"k8s.io/kubernetes/pkg/master"
+	"k8s.io/kubernetes/test/integration/framework"
+	netutils "k8s.io/utils/net"
+)
+
+// TestDualStackEnabledSingleIPv4Family tests a single IPv4 family API server with DualStack enabled
+func TestDualStackEnabledSingleIPv4Family(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.IPv6DualStack, true)()
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	clientSet, _ := framework.StartTestServer(t, stopCh, framework.TestServerSetup{
```

Comment:
after checking several integrations tests, seems there is no consistency on how to start a kube-apiserver, or I'm missing the goal of using one or another way, 
what's the recommended way of starting a kube-apiserver for these tests?

Reply:
for something like this where you want to exercise a variety of invocation options, and don't care about controller-manager running, I'd probably use

```go
import (
  kubeapiservertesting "k8s.io/kubernetes/cmd/kube-apiserver/app/testing"
  "k8s.io/kubernetes/test/integration/framework"
)

...

kubeapiservertesting.StartTestServerOrDie(t, nil, []string{
  ... apiserver flags go here ...
}, framework.SharedEtcd())
```

Reply:
that lets you explicitly enable/disable the feature with a `--feature-gates` arg, and explicitly set up an IPv4 or IPV6 or dual-stack API server configuration